### PR TITLE
Extract the state machine for WorkVersion to its own file

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔

The file was getting too long. This makes it easier to find what you are looking for.

## How was this change tested? 🤨
CI

### CODE REVIEW

**Review by ndushay — COMMENTED** [review]
Here, I feel like splitting out the state machine into a class ... that is only used by one class ... is just making me have to look in more files to find something.   If I'm in a minority on this point, so be it;  we can go with the majority.

But maybe it's better to just put the state machine at the bottom of the file.

Another argument against is that we're doing this for only ONE state machine?

More files doesn't always mean more readable -- having to jump between files to read code has a cost, too.

**Review by ndushay — APPROVED** [review]
approving, but wondering - when do we use concerns vs using a service?

**Comment by ndushay** [comment]
oh no - test failures!

File: app/models/collection_version.rb [hunk 1]
```diff
@@ -14,65 +14,6 @@ def accessioned?
     %w[first_draft depositing].exclude?(state)
   end
```
Comment (ndushay) [inline app/models/collection_version.rb:16]: is this missing include statement?

Comment (mjgiarlo) [reply app/models/collection_version.rb:16]: Sure looks like it.

Comment (jcoyne) [reply app/models/collection_version.rb:18]: Oops, I forgot to add that change to the commit. Got it now.

**Pull request merged**